src/libostree/ostree-sysroot-deploy.c

```diff
@@ -3360,6 +3407,9 @@ sysroot_finalize_deployment (OstreeSysroot *self, OstreeDeployment *deployment,
   if (!selinux_relabel_var_if_needed (self, sepolicy, os_deploy_dfd, cancellable, error))
     return FALSE;
 
+  if (!sysroot_initialize_deployment_backing (self, deployment, sepolicy, error))
```

Comment:
Shouldn't this be gated on `root.transient`?

Reply:
You're right to call this out, but this was semi-intentional because one thing I want to do is change `usroverlay` to write here instead of the `/var/tmp` it does today - because that would just be better; the content would be clearly "lifecycle bound" to the deployment and we wouldn't need that hacky `tmpfiles.d` snippet we carry:

```
# https://github.com/ostreedev/ostree/issues/393
R! /var/tmp/ostree-unlock-ovl.*
```

(That said to do that note we'd need to also change some logic to ensure the overlay content for it is actually removed on reboot by default)

This all said, yes we could still create it dynamically in either case.  I think we actually *should* have logic here in ostree to do what is being replicated in ostree-rs-ext and rpm-ostree around querying the initramfs config.  Then we could drop the `ex-integrity.composefs` repo config.